xl-v1.go

```diff
@@ -429,8 +429,26 @@ func (xl XL) StatFile(volume, path string) (FileInfo, error) {
 		return FileInfo{}, errInvalidArgument
 	}
 
+	readLock := true
```

Comment:
> The stale locks are differentiated by keeping them in a map and purging them subsequently based on the active count and understanding if they are InUse.

Yes, I understand now.. but won't that lead to unexpected behaviours somewhere else? in that case, you should make sure that routines which are waiting for the lock exit too with preserving the system state consistent